### DISCUSSION TITLE
Support pubsub and queue protocols (Kafka, Pulsar, AMQ, RMQ, MQTT, cloud-specific...)

### DISCUSSION THREAD

**Discussion by schrepfler** [testingisdocumenting/webtau#1069] [Ideas]
There's a whole set of test cases which are driven (producers) and verifiable (consumers) by pubsub type of integrations like Kafka, Pulsar, RabbitMQ, ActiveMQ etc. 
If we consider the table/stream dual, perhaps we can map current loaders to input streams (publishers) and then assert side-effects on the system by using the various webtau integrations (REST, DB but also other output streams etc.)
What's perhaps an open question how to capture each input to the step that needs to be asserted and then enforce this sequence.

This is one approach taken by the Weaver Test framework so take into account this is Scala https://github.com/disneystreaming/weaver-test

**Comment by MykolaGolubyev** [comment]
Hello,
I quite like it and it makes a lot of sense to me. Thank you for starting this discussion.
I am familiar with queues and their usage but their usage in my projects over the last years were close to 0. May help to feel the pain to get to the right solution. 
Wonder if extending the demo app that is part of repo with queues will be not a huge effort. 

Going to take a look at `weaver`. Thanks for sharing it.

Any chance you can share your preferred syntax/DSL for interactions?

**Comment by MykolaGolubyev** [comment]
@schrepfler wanted to check if you had a simple-ish use case in mind and a syntax proposal to collaborate on

**Reply by MykolaGolubyev** [reply]
@schrepfler ping?
